lib/AST/ArchetypeBuilder.cpp

```diff
@@ -2028,7 +2028,10 @@ Type ArchetypeBuilder::mapTypeIntoContext(Module *M,
   if (!canType->hasTypeParameter())
     return type;
 
-  assert(genericParams && "dependent type in non-generic context");
+  if (!genericParams) {
+    // Failed: dependent type in non-generic context.
+    return Type();
+  }
```

Comment:
I added these checks recently. There is a distinction between failure conditions that arise from invalid user input and failure conditions that arise from bugs inside the type checker. The latter should assert.

Note that if there are no generic parameters, we should not hit the assert, because the given type should not contain any type parameters. So if canType->hasTypeParameter() is true but genericParams is null, we have a bug.

Are you seeing this in one of the tests?